owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsPictureUploadsViewModelTest.kt

```diff
@@ -0,0 +1,286 @@
+        every { preferencesProvider.getString(any(), any()) } returns examplePath
+
+        picturesViewModel.loadPictureUploadsPath()
+        val uploadPath = picturesViewModel.getPictureUploadsPath()
+
+        assertEquals(examplePath, uploadPath)
+    }
+
+    @Test
+    fun `load picture uploads source path - ok`() {
+        mockkStatic(PreferenceManager.CameraUploadsConfiguration::class)
+
+        every { preferencesProvider.getString(any(), any()) } returns exampleSourcePath
+        // It has to be "" for the test to pass
+        every { PreferenceManager.CameraUploadsConfiguration.getDefaultSourcePath() } returns ""
+
+        picturesViewModel.loadPictureUploadsSourcePath()
+
+        verify(exactly = 1) {
+            preferencesProvider.getString(
+                PreferenceManager.PREF__CAMERA_PICTURE_UPLOADS_SOURCE,
+                PreferenceManager.CameraUploadsConfiguration.getDefaultSourcePath()
+            )
+        }
+    }
+
+    @Test
+    fun `get picture uploads source path - ko - no load before`() {
+        val uploadSourcePath = picturesViewModel.getPictureUploadsSourcePath()
+
+        assertNull(uploadSourcePath)
+    }
+
+    @Test
+    fun `get picture uploads source path - ok - load before`() {
+        mockkStatic(PreferenceManager.CameraUploadsConfiguration::class)
+
+        every { preferencesProvider.getString(any(), any()) } returns exampleSourcePath
+        every { PreferenceManager.CameraUploadsConfiguration.getDefaultSourcePath() } returns ""
+
+        picturesViewModel.loadPictureUploadsSourcePath()
+        val uploadSourcePath = picturesViewModel.getPictureUploadsSourcePath()
+
+        assertEquals(exampleSourcePath, uploadSourcePath)
+    }
+
+    @Test
+    fun `handle select picture uploads path - ok`() {
+        val data: Intent = mockk()
+        val ocFile: OCFile = mockk()
+
+        every { ocFile.remotePath } returns exampleRemotePath
+        every { data.getParcelableExtra<OCFile>(any()) } returns ocFile
+
+        picturesViewModel.handleSelectPictureUploadsPath(data)
+
+        verify(exactly = 1) {
+            data.getParcelableExtra<OCFile>(UploadPathActivity.EXTRA_FOLDER)
+            ocFile.remotePath
+            preferencesProvider.putString(PreferenceManager.PREF__CAMERA_PICTURE_UPLOADS_PATH, exampleRemotePath)
+        }
+    }
+
+    @Test(expected = NullPointerException::class)
+    fun `handle select picture uploads path - ko - data intent is null`() {
+        picturesViewModel.handleSelectPictureUploadsPath(null)
+    }
+
+    @Test(expected = NullPointerException::class)
+    fun `handle select picture uploads path - ko - folder to upload is null`() {
+        val data: Intent = mockk()
+
+        every { data.getParcelableExtra<OCFile>(any()) } returns null
+
+        picturesViewModel.handleSelectPictureUploadsPath(data)
+
+        verify(exactly = 1) {
+            data.getParcelableExtra<OCFile>(UploadPathActivity.EXTRA_FOLDER)
+        }
+    }
```

Comment:
Same as other tests. We should not expect a `NullPointerException`. We don't handle it. It will crash the app 🤯